File: src/022/p022.js
const fs = require('fs');

function loadNames() {
  const names = fs.readFileSync('p022_names.txt').toString();
  const namesArray = names.replace(/['"]+/g, '').split(',');
  const sortedNames = namesArray.sort((a, b) => {
    if (a < b) return -1;
    else if (a > b) return 1;
    return 0;
  });
  return sortedNames;
}

function score(name, index) {
  let nameScore = 0;
  for (let i = 0; i < name.length; i++) {
    nameScore += name.charCodeAt(i) - 64;
  }
  return nameScore * (index + 1);
}

function solve() {
  const sum = (acc, next) => acc + next;
  return loadNames().map(score).reduce(sum);
}

const result = solve();
console.log(result);

File: src/022/p022.rs
use std::fs::File;
use std::io::Read;

fn load_names() -> Vec<String> {
    let mut contents = String::new();
    File::open("p022_names.txt")
        .expect("File not found")
        .read_to_string(&mut contents)
        .expect("Error reading");

    let mut names = contents
                        .replace("\"", "")
                        .split(',')
                        .map(str::to_string)
                        .collect::<Vec<String>>();
    names.sort();
    names
}

fn name_score(pair: (usize, String)) -> i32 {
    let (i, name) = pair;
    let bytes: i32 = name.bytes().map(|c| (c as i32) - 64).sum();
    let index = (i + 1) as i32;
    bytes * index
}

fn solve() -> i32 {
    load_names()
        .into_iter()
        .enumerate()
        .map(name_score)
        .sum()
}

fn main() {
    let result = solve();
    println!("{}", result);
}
